core/src/main/scala/org/apache/spark/sql/delta/GeneratedColumn.scala

```diff
@@ -362,7 +362,19 @@ object GeneratedColumn extends DeltaLogging with AnalysisHelper {
                   createExpr(name)(SubstringPartitionExpr(partColName, pos, len))
                 case TruncTimestamp(
                   StringLiteral(format), ExtractBaseColumn(name, TimestampType), _) =>
-                    createExpr(name)(DateTruncPartitionExpr(format, partColName))
+                    createExpr(name)(TimestampTruncPartitionExpr(format, partColName))
+                case TruncTimestamp(
+                  StringLiteral(format),
+                  Cast(ExtractBaseColumn(name, DateType), TimestampType, _, _), _) =>
+                  format match {
```

Comment:
Why do we only support formats in the date-range? I know it wouldn't really make sense to trunc a date to "HOUR" for example, but I think our expression conversions should still hold? Let me know what you think

Reply:
You're right. While it makes no difference to truncate to hour or minute, it can be implemented and the filters should work perfectly fine. I removed the restrictions.